gordon_janitor_gcp/msg_generators.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Module to collect classes that transform different inputs into messages that
+can be passed from one plugin to another through a channel.
+
+Example A record message generator usage:
+
+.. code-block:: pycon
+
+    >>> import asyncio
+    >>> loop = asyncio.get_event_loop()
+    >>> instance1_data = {'hostname': 'a-host', 'internal_ip': '192.168.11.5'}
+    >>> instance2_data = {'hostname': 'b-host', 'internal_ip': '192.168.11.6'}
+    >>> generator = ARecordMsgGenerator('dnszone.com')
+    >>> batch = loop.run_until_complete(
+            generator.generate_batch([instance1_data, instance2_data]))
+    >>> print(batch)
+    {'zone': 'dnszone.com', 'rrsets':
+        [{'name': 'a-host', 'type': 'A', 'rrdatas': ['192.168.11.5']},
+         {'name': 'b-host', 'type': 'A', 'rrdatas': ['192.168.11.6']}]}
+"""
+
+
+class MsgGeneratorBase:
```

Comment:
The reason why I decided to cordon off message creation logic in its own module is that a lot more of it will come in order to handle our company specific record setup (multiple types of records, some business logic). 

The `ARecordMsgGenerator` is more of an example implementation than anything else since we won't use it.